build/configure/copy-helper.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+/**
+ * CopyHelper
+ */
+
+var utils = require('../utils');
+
+var CopyHelper = function(parentTarget, generator) {
+  this.parentTarget = parentTarget;
+  this.generator = generator;
+};
+
+CopyHelper.prototype = {
+
+  copy: function(src, dest) {
+    var srcFile = utils.getFile(src);
+
+    if (srcFile.isFile()) {
+      this.copyFile(src, dest);
+    } else if (srcFile.isDirectory()) {
+      this.copyDir(src, dest);
+    }
+  },
+
+  copyFile: function(src, dest) {
+    var cmd = '@mkdir -p ' + dest + '\n\t' + '@cp ' + src + ' $@';
+    var fullDest = utils.joinPath(dest, utils.basename(src));
+
+    this.generator.insertTask('', fullDest, [], [cmd]);
+    this.generator.insertDep(fullDest, src);
+    this.generator.insertDep(this.parentTarget, fullDest);
```

Comment:
We should prevent a case(I am not sure if it's common or not) 
```
copy(Apath, Bpath); // action 1
// modify something to B, then..
copy(Bpath, Cpath); // action 2
```
if we didn't set  action 2 depends on action 1,  we might probably get a makefile as below, 
```
all: Cpath Bpath

Cpath:
  @cp Bpath Cpath

Bpath:
  @cp Apath Bpath
```

We cannot guarantee target Bpath happens earlier than target Cpath especially in parallel.